### PR TITLE
chore: some CMakeLists and CMakePresets tweaks/fixes

### PR DESCRIPTION
**Pull request by ThirdEyeSqueegee**
Changes:
- Add `VCPKG_HOST_TRIPLET` to match `VCPKG_TARGET_TRIPLET`
- Add a few compiler and linker flags for debug/release configs for MSVC and clang-cl
- Suppress two warnings:
  - `C4996` (MSVC)
  - `-Wreturn-type` (clang-cl)
- Remove redundant `set(CMAKE_TOOLCHAIN_FILE ...)` call from CMakeLists.txt (it's already set in CMakePresets.json)
- Remove unused `clang` build preset and `CMAKE_C_COMPILER` cache variable

### CODE REVIEW

File: CommonLibSF/CMakeLists.txt [hunk 1]
```diff
@@ -14,7 +15,16 @@ set(CMAKE_CXX_STANDARD_REQUIRED ON)
...
+
+message("\nCMAKE_BUILD_TYPE: ${CMAKE_BUILD_TYPE}")
+message("\nCMAKE_CXX_FLAGS: ${CMAKE_CXX_FLAGS}")
+message("CMAKE_SHARED_LINKER_FLAGS: ${CMAKE_SHARED_LINKER_FLAGS}")
+
+message("\nCMAKE_CXX_FLAGS_RELEASE: ${CMAKE_CXX_FLAGS_RELEASE}")
+message("CMAKE_SHARED_LINKER_FLAGS_RELEASE: ${CMAKE_SHARED_LINKER_FLAGS_RELEASE}")
+
+message("\nCMAKE_CXX_FLAGS_DEBUG: ${CMAKE_CXX_FLAGS_DEBUG}")
+message("CMAKE_SHARED_LINKER_FLAGS_DEBUG: ${CMAKE_SHARED_LINKER_FLAGS_DEBUG}")
```
Comment (gottyduke) [inline CommonLibSF/CMakeLists.txt:27]: These are not needed, CMake already displays these variables right at the beginning when executing a preset.

**Comment by gottyduke** [comment]
Besides the redundant cmake statements, other changes look good to me. Just leave a reminder for the future similar changes:
`/OPT:REF` and `/OPT:ICF` is enabled by default outside Debug builds, so no need to explicitly pass it to Release configurations.

File: CommonLibSF/CMakeLists.txt (hunk 1 above)
Comment (ThirdEyeSqueegee) [reply CommonLibSF/CMakeLists.txt:27]: Ah I just added them bc it's hard to sift them out from all the other variables displayed at the beginning, I can remove them now that I'm done testing

**Comment by ThirdEyeSqueegee** [comment]
> Besides the redundant cmake statements, other changes look good to me. Just leave a reminder for the future similar changes: `/OPT:REF` and `/OPT:ICF` is enabled by default outside Debug builds, so no need to explicitly pass it to Release configurations.

Fosho I'll remove `/OPT:REF`, will leave `/OPT:ICF=4` since that needs to be specified explicitly to take effect

**Review by gottyduke — APPROVED** [review]

**Comment by ThirdEyeSqueegee** [comment]
@gottyduke necro-ing this bc I can't find any info about this online--how do you know which flags `CMAKE_INTERPROCEDURAL_OPTIMIZATION` sets?

**Pull request merged**